go/ethadapter/geth_rpc_client.go

```diff
@@ -8,37 +8,37 @@ import (
 	"math/big"
 	"time"
 
+	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
 	gethlog "github.com/ethereum/go-ethereum/log"
 
-	"github.com/ten-protocol/go-ten/contracts/generated/ManagementContract"
-	"github.com/ten-protocol/go-ten/go/common/retry"
-
-	"github.com/ten-protocol/go-ten/go/common/log"
+	lru "github.com/hashicorp/golang-lru/v2"
 
+	"github.com/ten-protocol/go-ten/contracts/generated/ManagementContract"
 	"github.com/ten-protocol/go-ten/go/common"
-
-	"github.com/ethereum/go-ethereum"
-
-	gethcommon "github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ten-protocol/go-ten/go/common/log"
+	"github.com/ten-protocol/go-ten/go/common/retry"
 )
 
 const (
 	connRetryMaxWait        = 10 * time.Minute // after this duration, we will stop retrying to connect and return the failure
 	connRetryInterval       = 500 * time.Millisecond
 	_maxRetryPriceIncreases = 5
 	_retryPriceMultiplier   = 1.2
+	_defaultBlockCacheSize  = 51 // enough for 50 request batch size and one for previous block
```

Comment:
This is just a finger in the air placeholder, I tried it with 200 and it was fine but doesn't give you many more cache hits so I went for a reasonable minimum.

We can make this configurable at some point if it ever becomes useful.